app/components/build-count.js

```diff
@@ -0,0 +1,94 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import { pluralize } from 'ember-inflector';
+import { reads, equal, or } from '@ember/object/computed';
+import { task } from 'ember-concurrency';
+import { DEFAULT_INSIGHTS_INTERVAL } from 'travis/services/insights';
+
+export default Component.extend({
+  classNames: ['insights-build-count'],
+  private: false,
+  interval: DEFAULT_INSIGHTS_INTERVAL,
+  owner: null,
+
+  insights: service(),
+
+  // Current Interval Chart Data
+  requestData: task(function* () {
+    return yield this.insights.getChartData.perform(
+      this.owner,
+      this.interval,
+      'builds',
+      'sum',
+      ['count_started'],
+      {
+        calcAvg: true,
+        private: this.private,
+      }
+    );
+  }),
+  chartData: reads('requestData.lastSuccessful.value'),
+  builds: reads('chartData.data.count_started.plotValues'),
+  labels: reads('chartData.labels'),
+
+  isLoading: reads('requestData.isRunning'),
+  isEmpty: equal('totalBuilds', 0),
+  showPlaceholder: or('isLoading', 'isEmpty'),
+
+  // Total / average
+  totalBuilds: reads('chartData.data.total'),
+  avgBuilds: reads('chartData.data.average'),
+
+  totalBuildText: computed('totalBuilds', function () {
+    if (typeof this.totalBuilds !== 'number') { return '\xa0'; }
+    return this.totalBuilds.toLocaleString();
+  }),
+
+  // Previous interval chart data
+  requestPastData: task(function* () {
+    return yield this.insights.getChartData.perform(
+      this.owner,
+      this.interval,
+      'builds',
+      'sum',
+      ['count_started'],
+      { startInterval: -2, endInterval: -1, calcTotal: true, calcAvg: true, private: this.private }
+    );
+  }),
```

Comment:
I would suggest to add `drop` modifier to this task, because it's getting called whenever component arguments change, which may happen often, and in this case, it will hit API multiple times for the same data. Here's the link that explains how concurrency modifiers work: http://ember-concurrency.com/docs/task-concurrency

Reply:
Nice, that makes sense. Added 😄